xarray/backends/zarr.py

```diff
@@ -409,7 +430,10 @@ def ds(self):
 
     def open_store_variable(self, name, zarr_array):
         data = indexing.LazilyIndexedArray(ZarrArrayWrapper(name, self))
-        dimensions, attributes = _get_zarr_dims_and_attrs(zarr_array, DIMENSION_KEY)
+        try_nczarr = self._mode not in ["a", "r+"]
```

Comment:
I wonder, would `self._mode == 'r'` be clearer here?

Reply:
Yes, clearer and achieve the same!